### PR TITLE
[ENH] events.tsv should be sorted by onset values

### LINKED ISSUE #1691: Event onset times can be out of order
Some datasets on OpenNeuro do not have their events ordered in ascending order by the `onset` time. I couldn't find anything in the spec about this.... but I think downstream analysts have a reasonable expectation that event times should be non-decreasing and will be in for a nasty surprise (if they notice it) when using these datasets.

Should there be a requirement that time series times be non-decreasing?

### PR DESCRIPTION
**Pull request by Remi-Gau**
- closes #1691

### CODE REVIEW

**Comment by effigies** [comment]
Will propose schema rule.

**Review by sappelhoff — APPROVED** [review]

File: src/schema/rules/checks/events.yaml [hunk 1]
```diff
@@ -27,3 +27,15 @@ StimulusFileMissing:
...
+  issue:
+    code: EVENT_ONSET_ORDER
+    message: |
+      The onset column in events.tsv files should be sorted.
+    level: warning
+  selectors:
+    - suffix == "events"
+    - extension == ".tsv"
+  checks:
+    - sorted(columns.onset) == columns.onset
```
Comment (effigies) [inline src/schema/rules/checks/events.yaml:41]: @rwblair Do you agree?

Comment (rwblair) [reply src/schema/rules/checks/events.yaml:41]: n/a would be the only thing to mess it up if allowed. Otherwise looks good.

Comment (effigies) [reply src/schema/rules/checks/events.yaml:41]: Good point. I think it being a warning should make that an annoyance, not a show-stopper. Will add a comment, though.

Comment (effigies) [inline src/schema/rules/checks/events.yaml:41]:
```suggestion
    # n/a values will likely cause false alarms if encountered. Consider alternatives.
    - sorted(columns.onset) == columns.onset
```

**Review by effigies — APPROVED** [review]

**Pull request merged**